### PR TITLE
Dev iothub microsoft.devices 2021 07 15 preview

### PR DESCRIPTION
**Pull request by Rickyp**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by zhenglaizhang** [comment]
Hi @Rickyp The PR is pending on ARM review, if you are feeling blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

**Comment by j5lim** [comment]
>         },

Please annotate them with x-ms-secret. Also please make sure they are not returned on GET. Secrets can be returned only via POST.

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

---
Refers to: specification/iothub/resource-manager/Microsoft.Devices/preview/2021-07-15-preview/iothub.json:3192 in 9b7ed32. [](commit_id = 9b7ed32c70aae6a1eff7e1f0d0b8a4f6db50e098, deletion_comment = False)

**Comment by j5lim** [comment]
>           "type": "string"

Is this a secret? If so, need to set x-ms-secret to true and not expose via GET.

---
Refers to: specification/iothub/resource-manager/Microsoft.Devices/preview/2021-07-15-preview/iothub.json:2214 in 9b7ed32. [](commit_id = 9b7ed32c70aae6a1eff7e1f0d0b8a4f6db50e098, deletion_comment = False)

**Comment by j5lim** [comment]
>           "description": "The connection string of the service bus queue endpoint.",

secret

---
Refers to: specification/iothub/resource-manager/Microsoft.Devices/preview/2021-07-15-preview/iothub.json:2938 in 9b7ed32. [](commit_id = 9b7ed32c70aae6a1eff7e1f0d0b8a4f6db50e098, deletion_comment = False)

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2021-07-15-preview/examples/iothub_certificatescreateorupdate.json
Comment (j5lim) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2021-07-15-preview/examples/iothub_certificatescreateorupdate.json:24]: >"############################################" [](http://example.com/codeflow?start=25&length=46)

Recommend not returning it in the response.

**Comment by j5lim** [comment]
>         "summary": "Get the certificate.",

Get cannot return a secret. It should be POST.

---
Refers to: specification/iothub/resource-manager/Microsoft.Devices/preview/2021-07-15-preview/iothub.json:1386 in 9b7ed32. [](commit_id = 9b7ed32c70aae6a1eff7e1f0d0b8a4f6db50e098, deletion_comment = False)

**Review by j5lim — APPROVED** [review]
Approved with suggestions from ARM side. The service team created a WI to address the unresolved comments.

**Comment by zhenglaizhang** [comment]
@Rickyp the PR LGTM, and is good to merge, so I am checking with you if this api version has been deployed to PROD and ready to merge this PR? If so I will proceed to merge the PR

**Comment by Rickyp** [comment]
> @Rickyp the PR LGTM, and is good to merge, so I am checking with you if this api version has been deployed to PROD and ready to merge this PR? If so I will proceed to merge the PR

**Comment by Rickyp** [comment]
@zhenglaizhang , we are still pending to deploy the release where the new bits live for this swagger change, but I will circle back once that is confirmed if that is okay with you?

**Comment by zhenglaizhang** [comment]
@Rickyp sure, I added `DoNotMerge` label and when you are ready to merge, just remove the label to let me know

**Pull request closed without merging**
